src/sentry/sentry_metrics/indexer/ratelimiters.py

```diff
@@ -139,7 +144,9 @@ def check_write_limits(self, use_case_id: UseCaseKey, keys: KeyCollection) -> Ra
         """
 
         org_ids, requests = _construct_quota_requests(use_case_id, keys)
-        timestamp, grants = self._get_rate_limiter(use_case_id).check_within_quotas(requests)
+        timestamp, grants = self._get_rate_limiter(use_case_id, indexer_db).check_within_quotas(
```

Comment:
How does the rate limiter namespacing work? Looking at sliding_windows.py the redis key and cluster seems to be the same.

I think the rate limiter should not need to get passed the indexer_db every time or need to know anything about the storage backends for that matter, just have a different namespace.

Reply:
I think that is what @fpacifici  meant by having the rate limiting partition be a separate attribute in the metrics configuration. I think im gonna open a separate PR to do that for the rate limiter since that can be independent from the backends

Reply:
I think making this work will be more complex than just passing the indexer_db.
The Configuration object references a structure in server.py which contains the redis connection settings. 
If the test indexer worked on a different cluster that would be enough, But this is unlikely. We will likely share the cluster so we need to implement a way in the rate limiter initialization to provide a prefix in the key as well and to configure that via settings.

Reply:
More context: 
- now each instance of the indexer creates a number of rate limiters objects (one per use case id). The configuration for each of them in provided by the `Configuration` object. So we can have multiple of them in the same indexer per use case id. Each can reference a different cluster .
- Name spacing in the rate limiter is defined via the `Quota` object per use case id. So the use case id ends up in the key thus guaranteeing the name spacing https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_metrics/indexer/ratelimiters.py#L33-L58
- So we have a mix of rate limiter configuration set at initialization and of parameters passed when the rate limiter is invoked, which is suboptimal. This most likely happened because the `Configuration` abstraction was added at the same time as the rate limiter was being developed.

So there is some refactoring needed here to solve your use case. 
- we cannot rely on the db_id alone and hope that the rate limiter configuration we get from the `Configuration` object will give us a separate name space. That config only contains the cluster to connect to. Which is likely going to be the same.
- we cannot rely on the use case id to be different. It will certainly be the same. 

So there definitively need to be something to add to the `Configuration` object to identify the rate limiter namespace. 
We can temporarily use the db id for this PR but we should have a separate one as I mentioned in my comment last week. 
Then, this string that comes from the config needs to be added to the redis keys of our rate limiter. We will have two options:
- Adding the prefix to the quota or request quota here. You can add a prefix override there. https://github.com/getsentry/sentry/blob/master/src/sentry/sentry_metrics/indexer/ratelimiters.py#L33-L80
- make changes to the basic rate limiter implementation to add a prefix at initialization. I think this refactoring is beyond the scope as all prefix overrides are already defined as part of the Quota or RequestQuota object and not at initialization.